org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/ScalaIndenter.scala

```diff
@@ -1518,10 +1517,9 @@ class ScalaIndenter(
    */
   private def isStringOrCharLiteralAssignment(referenceTokenPos: Int, offset: Int) = {
     val restOfTheLine = document.get(referenceTokenPos, Math.max(offset - referenceTokenPos, 0))
-    val charLit = """'(.|\\\w|\\\\)'"""
-    val stringLit = """"(\\"|[^"])*")"""
-
-    val regex = s"(${charLit}|${stringLit}$$".r
+    val charLit = "'.'"
+    val stringLit = "\".*\""
```

Comment:
I still think you need to accept escaped strings. This regex does not match `"quote inside \""`.